beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithInArrearSubscriptions.java

```diff
@@ -150,6 +153,320 @@ public void testWithPauseResume() throws Exception {
+        final PlanPhaseSpecifier spec = new PlanPhaseSpecifier("premium-support-monthly-notrial", null);
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.NULL_INVOICE);
+        final UUID bpEntitlementId = entitlementApi.createBaseEntitlement(account.getId(), new DefaultEntitlementSpecifier(spec, null, null, null), "bundleExternalKey", null, null, false, true, ImmutableList.<PluginProperty>of(), callContext);
+        assertListenerStatus();
+        Entitlement bpEntitlement = entitlementApi.getEntitlementForId(bpEntitlementId, callContext);
+        assertNotNull(bpEntitlement);
+        assertEquals(bpEntitlement.getState(), EntitlementState.ACTIVE);
+        assertEquals(bpEntitlement.getLastActivePlan().getName(), "premium-support-monthly-notrial");
+        checkNoMoreInvoiceToGenerate(account);
+
+        //CREATE ADDON SUBSCRIPTION. SINCE IN_ARREAR BILLING IS USED, INVOICE IS NOT GENERATED
+        final PlanPhaseSpecifier addOnSpec = new PlanPhaseSpecifier("premium-support-addon-monthly-notrial", null);
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.NULL_INVOICE);
+        final UUID addonEntitlementId = entitlementApi.addEntitlement(bpEntitlement.getBundleId(), new DefaultEntitlementSpecifier(addOnSpec), today, today, false, ImmutableList.<PluginProperty>of(), callContext);
+        Entitlement addOnEntitlement = entitlementApi.getEntitlementForId(addonEntitlementId, callContext);
+        assertEquals(addOnEntitlement.getState(), EntitlementState.ACTIVE);
+        assertEquals(addOnEntitlement.getLastActivePlan().getName(), "premium-support-addon-monthly-notrial");
+        assertListenerStatus();
+        checkNoMoreInvoiceToGenerate(account);
+
+        //MOVE CLOCK
+        clock.addDays(10);
+
+        //CHANGE PLAN WITH START_OF_TERM POLICY
+        final PlanPhaseSpecifier newPlanSpec = new PlanPhaseSpecifier("basic-support-monthly-notrial", null);
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.CANCEL, NextEvent.BLOCK, NextEvent.NULL_INVOICE, NextEvent.NULL_INVOICE); //TODO: should this be CHANGE instead of CREATE? Also, need to check if other events are correct once bug is fixed
+        bpEntitlement.changePlanOverrideBillingPolicy(new DefaultEntitlementSpecifier(newPlanSpec), null, BillingActionPolicy.START_OF_TERM, null, callContext);
+        assertListenerStatus();
+        bpEntitlement = entitlementApi.getEntitlementForId(bpEntitlement.getId(), callContext);
+        assertEquals(bpEntitlement.getState(), EntitlementState.ACTIVE);
+        assertEquals(bpEntitlement.getLastActivePlan().getName(), "basic-support-monthly-notrial");
+        addOnEntitlement = entitlementApi.getEntitlementForId(addonEntitlementId, callContext);
+        assertEquals(addOnEntitlement.getState(), EntitlementState.CANCELLED);
+        checkNoMoreInvoiceToGenerate(account);
+
+    }
+
+    @Test(groups = "slow", enabled = false)
```

Comment:
Is this one for `https://github.com/killbill/killbill/issues/1630` and is this why you have disabled it?